executor.go

```diff
@@ -176,23 +190,49 @@ func (e *Executor) verifySubjectAgainstReferrers(ctx context.Context, task *exec
 		}
 		return nil
 	})
-	if err != nil {
+	if err != nil && err != ErrSubjectPruned {
 		return nil, fmt.Errorf("failed to verify referrers for artifact %s: %w", artifact, err)
 	}
+	if evaluator != nil {
+		if err := evaluator.Commit(ctx, task.artifactDesc.Digest.String()); err != nil {
+			return nil, fmt.Errorf("failed to evaluate artifact %s: %w", artifact, err)
+		}
+	}
 	task.subjectReport.ArtifactReports = append(task.subjectReport.ArtifactReports, artifactReports...)
 
+	if err == ErrSubjectPruned {
+		return nil, nil
+	}
 	return newTasks, nil
 }
 
 // verifyArtifact verifies the artifact by all configured verifiers and returns
 // error if any of the verifier fails.
-func (e *Executor) verifyArtifact(ctx context.Context, repo string, subjectDesc, artifact ocispec.Descriptor) ([]*VerificationResult, error) {
+func (e *Executor) verifyArtifact(ctx context.Context, repo string, subjectDesc, artifact ocispec.Descriptor, evaluator Evaluator) ([]*VerificationResult, error) {
 	var verifierReports []*VerificationResult
 
 	for _, verifier := range e.Verifiers {
 		if !verifier.Verifiable(artifact) {
 			continue
 		}
+
+		if evaluator != nil {
+			prunedState, err := evaluator.Pruned(ctx, subjectDesc.Digest.String(), artifact.Digest.String(), verifier.Name())
+			if err != nil {
+				return nil, fmt.Errorf("failed to check if verifier %s is required to verify subject %s against artifact %s, %w", verifier.Name(), subjectDesc.Digest.String(), artifact.Digest.String(), err)
+			}
+			switch prunedState {
+			case PrunedStateVerifierPruned:
+				continue
+			case PrunedStateArtifactPruned:
+				return verifierReports, nil
```

Comment:
nit: just for readability
```suggestion
				break
```